lib/parse.js

```diff
@@ -396,8 +422,10 @@ define(['esprima', './lang'], function (esprima, lang) {
             if (funcArgLength) {
                 deps = (funcArgLength > 1 ? ["require", "exports", "module"] :
                         ["require"]).concat(deps);
+                params = getFormalParameters(node).concat(params);
```

Comment:
Just a note for me to check this more when I get back to the right computer.

Reply:
It concatenates formal parameter names from the function prototype with variable names from the module callback body. If it is pure AMD, usually only the former array is filled. If it is a CJS-like AMD wrapper, the former array contains just the pseudo-modules and the latter array contains the  rest of dependency names.